### ISSUE TITLE
Passing TData causes IPagination to never return ILengthAwarePagination

### ISSUE THREAD

**Issue by BrettBedarf** [felixmosh/knex-paginate#19]
```  typescript

// works

const result = await db('users').select('*').paginate({
        perPage: 10,
        currentPage: 1,
        isLengthAware:true,
      })
    );

result.pagination.total // ✅

// doesn't work
const result = await db('users').select('*').paginate<User[]>({
        perPage: 10,
        currentPage: 1,
        isLengthAware:true,
      })
    );

result.pagination.total  // ❌ Property 'total' does not exist on type 'IBasePagination'.
```

**Comment by felixmosh** [comment]
Thank you for reporting this issue 🙏🏼
I'll look into it asap

**Comment by felixmosh** [comment]
Released in `v2.3.0`
Now it works with the original data type of Knex.

```js
const result = await db<User[]>('users').select('*').paginate({
  perPage: 10,
  currentPage: 1,
  isLengthAware: true,
});
```
